packages/widget/src/hooks/useChains.ts

```diff
@@ -11,21 +11,32 @@ export const useChains = () => {
   const initializeChains = useChainOrderStore(
     (state) => state.initializeChains,
   );
+
   const { data: availableChains, isLoading: isLoadingAvailableChains } =
     useQuery(['chains'], async () => lifi.getChains(), {
       refetchInterval: 300000,
       staleTime: 300000,
     });
 
   const { data: filteredChains, isLoading: isLoadingFilteredChains } = useQuery(
-    ['filtered-chains', availableChains?.length, keyPrefix],
+    ['filtered-chains', { keyPrefix, availableChains, chains }],
     async () => {
       if (!availableChains) {
         return;
       }
       const filteredChains = availableChains.filter((chain) =>
         isItemAllowed(chain.id, chains),
       );
+
+      return filteredChains;
+    },
+    {
+      enabled: Boolean(availableChains),
+    },
+  );
+
+  useEffect(() => {
```

Comment:
So had a problem in the UI where it there is an attempt to reset the chains to the default settings of an empty array or undefined that the following UI can happen where there are 4 missing chains (there should be the full 18 in teh image below).

<img width="530" alt="issue-rendering chains" src="https://github.com/lifinance/widget/assets/7836726/55bfed1c-b1a5-483f-a95f-b3fa7b493890">

Only way I've found so far to get around this is to use a useEffect block to make sure the zustand. Going to see if there is a function on React Queries useQuery hook that might work as well

Reply:
I'm not finding any ting in relation to useQuery thats going to be helpful and I the useEffect will hav to stay if we want to avoid the problem pictured here